kptncook/mealie.py

```diff
@@ -187,51 +172,265 @@ def login(self, username: str = "admin", password: str = ""):
+        r.close()
+        download_file.close()
+        r.raise_for_status()
+
+        return r.json()["fileName"]
+
+    def _build_recipestep_text(self, recipe_uuid, text, image_name):
+        return f"{text} <img src=\"/api/media/recipes/{recipe_uuid}/assets/{image_name}\" height=\"100%\" width=\"100%\"/>"
+
+    def enrich_recipe_with_step_images(self, recipe):
+        for i in range(len(recipe.recipeInstructions)):
+            uploaded_image_name = self.upload_asset(recipe.slug, recipe.recipeInstructions[i].image)
+            recipe.recipeInstructions[i].text = self._build_recipestep_text(recipe.id, recipe.recipeInstructions[i].text, uploaded_image_name)
+
+        return recipe
+
+
+    def _post_recipe_trunk_and_get_slug(self, recipe_name):
+        data = {"name": recipe_name}
+        r = self.post("/recipes", data=json.dumps(data))
+        r.close()
         r.raise_for_status()
         slug = r.json()
         return slug
 
     def _scrape_image_for_recipe(self, recipe, slug):
-        json_image_url = json.dumps({"url": recipe.image_url})
+        json_image_url = json.dumps({"url": recipe.imageUrl})
         scrape_image_path = f"/recipes/{slug}/image"
         r = self.post(scrape_image_path, data=json_image_url)
+        r.close()
         r.raise_for_status()
 
     def _update_user_and_group_id(self, recipe, slug):
         recipe_detail_path = f"/recipes/{slug}"
         r = self.get(recipe_detail_path)
+        r.close()
         r.raise_for_status()
         recipe_details = r.json()
         update_attributes = ["id", "userId", "groupId"]
         updated_details = {k: recipe_details[k] for k in update_attributes}
         recipe = RecipeWithImage(**(recipe.dict() | updated_details))
+        #for tag in recipe.tags:
+        #    tag.groupId = recipe_details["groupId"]
         return recipe
 
     def _update_recipe(self, recipe, slug):
         recipe_detail_path = f"/recipes/{slug}"
         r = self.put(recipe_detail_path, data=recipe.json())
+        r.close()
         r.raise_for_status()
         return Recipe.parse_obj(r.json())
 
     def create_recipe(self, recipe):
-        slug = self._post_recipe_and_get_slug(recipe)
+        slug = self._post_recipe_trunk_and_get_slug(recipe.name)
+        print(slug)
+        recipe.slug = slug
         self._scrape_image_for_recipe(recipe, slug)
         recipe = self._update_user_and_group_id(recipe, slug)
+        recipe = self.enrich_recipe_with_step_images(recipe)
         return self._update_recipe(recipe, slug)
 
     def get_all_recipes(self):
-        r = self.get("/recipes")
+        all_recipes = []
+
+        r = self.get("/recipes?page=1&perPage=50")
+        r.close()
         r.raise_for_status()
-        return [Recipe.parse_obj(recipe) for recipe in r.json()]
+        all_recipes.extend([Recipe.parse_obj(recipe) for recipe in r.json()["items"]])
+
+        page = 2
+        while page <= r.json()["total_pages"]:
```

Comment:
Well, I guess get("/recipes") didn't get all? 😬

Reply:
Nope and iirc the api call is already paged per default to 50. Additionally, for larger mealie instances it take a considerable time to get an answer from mealie, so I left it with a pagesize of 50.